From: Werner Koch <wk@gnupg.org>
Date: Fri, 29 Dec 2017 20:18:20 +0100
Subject: gpg: Allow the use of "cv25519" and "ed25519" in the keygen parms.

* g10/keygen.c (gen_ecc): Map curve names.
--

See
https://lists.gnupg.org/pipermail/gnupg-users/2017-December/059619.html

Signed-off-by: Werner Koch <wk@gnupg.org>
(cherry picked from commit 412bb7a801f242d47a82712080cce6ddbb843166)
---
 g10/keygen.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/g10/keygen.c b/g10/keygen.c
index 4354e92..77d1b84 100644
--- a/g10/keygen.c
+++ b/g10/keygen.c
@@ -1580,6 +1580,13 @@ gen_ecc (int algo, const char *curve, kbnode_t pub_root,
   if (!curve || !*curve)
     return gpg_error (GPG_ERR_UNKNOWN_CURVE);
 
+  /* Map the displayed short forms of some curves to their canonical
+   * names. */
+  if (!ascii_strcasecmp (curve, "cv25519"))
+    curve = "Curve25519";
+  else if (!ascii_strcasecmp (curve, "ed25519"))
+    curve = "Ed25519";
+
   /* Note that we use the "comp" flag with EdDSA to request the use of
      a 0x40 compression prefix octet.  */
   if (algo == PUBKEY_ALGO_EDDSA)
